src/librustc_mir/interpret/place.rs

```diff
@@ -1077,9 +1074,6 @@ where
                 discr_index,
                 ..
             } => {
-                if !variant_index.as_usize() < dest.layout.ty.ty_adt_def().unwrap().variants.len() {
-                    throw_ub!(InvalidDiscriminant(variant_scalar));
-                }
```

Comment:
Same, please keep that check.

Reply:
These checks are both impossible to trip (`for_variant` above or anywhere else would cause an out of bounds panic) and not even consistent with eachother (presumably `variant_range` is the "more correct" one).

Note that this is `VariantIdx` and *not* some arbitrary integer. It's more akin to a `DefId` in its expected validity, except cheaper to work with.

Reply:
`VariantIdx` is pretty easy to get from a `u32` so that doesn't really reassure me.

We mostly have these because it seems inconsistent to check for `Single` and not check for `Multiple`.

Reply:
`Single` has a pre-selected variant, hence the check (which codegen also has), but `Multiple` holds *all* the variants, so any of them is a valid index.

> `VariantIdx` is pretty easy to get from a `u32` so that doesn't really reassure me.

Okay (well that applies to any ID in the compiler, but that's besides the point) but `for_variant` would still panic here:
https://github.com/rust-lang/rust/blob/2da942f32802c8233a09744024dfbc34431adf65/src/librustc/ty/layout.rs#L2063-L2065

Going through *some other* information to check the `VariantIdx` doesn't make much sense to me.

Reply:
I'm always a fan of double-checking such things (but I agree the inconsistency between the two checks should be fixed).

However, I am also fine replacing them by a comment saying that `for_variant` already checked the variants existence.